js/src/lib/interfaces/Types.ts

```diff
@@ -758,88 +760,88 @@ export interface TransformersInfo {
  * This mapping is generated automatically by "python-api-export-tasks" action in huggingface/api-inference-community repo upon merge.
  * Ref: https://github.com/huggingface/api-inference-community/pull/158
  */
-export const LIBRARY_TASK_MAPPING_EXCLUDING_TRANSFORMERS: Record<string, Array<string>> = {
-	"adapter_transformers": [
+export const LIBRARY_TASK_MAPPING_EXCLUDING_TRANSFORMERS: Partial<Record<keyof typeof ModelLibrary, PipelineType[]>> = {
+	"adapter-transformers": [
 		"question-answering",
 		"text-classification",
-		"token-classification"
+		"token-classification",
 	],
 	"allennlp": [
-		"question-answering"
+		"question-answering",
 	],
 	"asteroid": [
-		"audio-source-separation",
-		"audio-to-audio"
+		// "audio-source-separation",
+		"audio-to-audio",
 	],
 	"diffusers": [
-		"text-to-image"
-	],
-	"doctr": [
-		"object-detection"
+		"text-to-image",
 	],
+	// "doctr": [
+	// 	"object-detection",
+	// ],
```

Comment:
Commented because it's absent from `ModelLibrary`

Reply:
Same for superb & k2_sherpa

Reply:
Should it be handled from the source huggingface/api-inference-community? (see  https://github.com/huggingface/api-inference-community/pull/158)

Reply:
@osanseviero Any thoughts how to handle the discrepancy. Is the inference API working for doctr, superb & k2_sherpa? Or do we need to add these to `ModelLibrary`?

I didn't find anything related to k2_sherpa in the hub. I had a quick glimpse on a few superb models and they seem to be using transformer for inference API (?) Most doctr repos are dummy models and the inference API is unable to decide the pipeline_tag.

Reply:
> Is the inference API working for doctr, superb & k2_sherpa? Or do we need to add these to ModelLibrary?

* Superb repos use the Community Inference API, but given they are not a library and their usage is more for a competition, they were not added to the `Libraries.ts` (confirmed with @lewtun). From a discussion, we think https://huggingface.co/models?library=superb&sort=modified are mostly old legacy models and it's ok to remove them from the Inference API, as the official ones now use `transformers`. (the old integration allowed users to specify their own requirements and inference code but that was deprecated)
*  I don't have context in the `k2-sherpa` integration, so maybe makes sense to remove entirely from the community API or exclude from the exporter
* https://huggingface.co/models?library=doctr&sort=modified is just one model so I'm fine removing `doctr`

Reply:
Agree that it's OK to remove all the models listed here: https://huggingface.co/models?library=superb&sort=modified

(These were just for testing)

Reply:
@xianbaoqian would you be up for opening a PR to exclude these? :fire:

Reply:
Sure. Will update [api-inference-community](https://github.com/huggingface/api-inference-community) to remove these.

Reply:
I'm a bit surprised that using library: k2 actually made the inference API working. I thought that it needs the same name as the folder name in api-inference-community. 

Can anyone share a pointer on how they communicate? Not sure if it's easier to just update the repo library name from k2 to k2_sherpa, or to rename k2_sherpa folder to k2. Either way, we also need to update the library constant in the .ts file.

Reply:
The `ModelInfo` `library_name` is not enforced/validated, so it can be anything. When sending requests to the Inference API, it will pick the library from this internal list (https://github.com/huggingface/api-inference/blob/main/master/app/types.py#L21), which, as you can see, contains some libraries that are not in `Libraries.ts` such as `doctr`, `superb`, `generic`, etc. 

**We do want a mechanism in which we want to run Inference API on new libraries without adding a filter at the left of hf.co/models though. Making this strongly typed would disallow us from doing so. cc @SBrandeis any ideas on this? **

Reply:
> We do want a mechanism in which we want to run Inference API on new libraries without adding a filter at the left of hf.co/models though.

Thanks for the information @osanseviero  I think both k2 and doctr could be good examples. They are only limited number of models available so it makes sense not showing them on the UI in the models page but it's probably good to support them on the inference API given the efforts that we put in to support them in the first place. (also cc @Narsil who has been working on k2 integration)

I do like strong typed list over weak typed list for many reasons such as detecting typos much early on, etc. Two thoughts:
- Maybe we can just create a different constant list of keys for libraries that supports inference API.
- or we can follow the example in https://github.com/huggingface/hub-docs/blob/main/js/src/lib/interfaces/Types.ts#L609 and defines a `hideInModels` flag for libraries. If this flag is on, the library will not show up in filter on the page to the models page. (kudos to @osanseviero)

Either way, this issue is getting more complicated. Probably we want to move forward with https://github.com/huggingface/moon-landing/pull/4816 first before the strong typed thing is figured out?

Any thoughts @SBrandeis @julien-c ?

Reply:
All up for static typing ! 

I think the current issues are not worth stopping the current PR given the quality of life improvement static typing provides.

That being said, the fact that the mapping is not easily aligned currently just shows to me that `doctr` and `k2` are just missing from the library keys, no ? So instead of removing them couldn't we add them where they are missing ?

If the issue would be making them visible as filters, then maybe we're just lacking an exclusion list from the filter in the search.

Reply:
IMO the simplest way is add `k2` and `doctr` (and any other future library implemented in api-inference) to the `ModelLibrary` type, but add a blocklist (or allow-list) of libraries that are not displayed in /models

Let me push a proposal

Reply:
64e8088c19cc80de897a3eb7393507db54e77f60

Reply:
for me this PR is ready to merge

Reply:
Thanks! I think adding the two definitely makes sense. 

Another issue is that people have been using k2 in HuggingFace tags and k2_sherpa in api-inference-community. I think the easiest fix is to rename k2_sherpa to k2. I'll propose a PR in api-inference-community!

Reply:
(also see internal PR https://github.com/huggingface/moon-landing/commit/e9557f8c848abbb98e48e8c4023ac6526022273e)

Reply:
and yes renaming k2_sherpa to k2 sounds perfect

Reply:
64e8088 was exactly what I had in mind when we discussed @xianbaoqian. :) Thanks for doing it @julien-c!